tests/sqllogictests/suites/base/03_common/03_0025_delete_from

```diff
@@ -264,9 +264,137 @@ select * from t order by c;
 
 
 
+
 statement ok
 drop table t all
 
+####################################
+# delete pruning, whole segments   #
+####################################
+
 statement ok
-DROP DATABASE db1
+create table t (c int);
+
+# add the first segment
+statement ok
+insert into t values(1),(2),(3);
+
+# add the second segment
+statement ok
+insert into t values(4),(5),(6);
+
+# add the third segment
+statement ok
+insert into t values(7),(8),(9);
+
+#clear metrics
+statement ok
+truncate table system.metrics;
+
+statement ok
+delete from t where c > 3 and c < 8;
+
+# expects 1 block pruned and 
+query I
+select value from system.metrics where metric = 'fuse_deletion_block_range_pruned_nums';
+----
+1.0
+
+query I
+select value from system.metrics where metric = 'fuse_deletion_segment_range_pruned_whole_segment_nums';
+----
+1.0
+
+query I
+select * from t order by c;
+----
+1
+2
+3
+8
+9
+
+# expects 1 whole block deletion: the block of value 2
+query I
+select value from system.metrics where metric = 'fuse_deletion_block_range_pruned_whole_block_nums';
+----
+1.0
+
+statement ok
+drop table t all
+
+# test large data
+statement ok
+create table t (c int) row_per_block = 100000 block_per_segment = 1000;
+
+statement ok
+set global max_threads = 1;
```

Comment:
will a session-level setting also meet the pre-conditions of the test setup?  other or later test cases may be "astonished" by this implicit setting

Reply:
will do it in next pr